dapp/src/components/LoadingModal.tsx

```diff
@@ -0,0 +1,26 @@
+import React from "react"
+import { ModalBody } from "@chakra-ui/react"
+import { TextMd } from "#/components/shared/Typography"
+import withBaseModal from "./ModalRoot/withBaseModal"
+import Spinner from "./shared/Spinner"
+
+export function LoadingModalBase() {
+  return (
+    <ModalBody>
+      <Spinner
+        size="2xl"
+        variant="filled"
+        borderTopColor="gold.100"
+        borderBottomColor="gold.100"
+        borderLeftColor="gold.100"
+      />
+      <TextMd fontWeight="semibold">Loading...</TextMd>
```

Comment:
See comment from above. We can render the `Spiner` and `Text` components in the `GateModal` component depending on the `isLoading` value from the `useQuery` hook.

Reply:
https://github.com/thesis/acre/pull/767#discussion_r1814757634